tapfreighter/wallet.go

```diff
@@ -1231,28 +1211,139 @@ func removeActiveCommitments(inputCommitment *commitment.TapCommitment,
 
 	// Remove input assets (the assets being spent) from list of assets to
 	// re-sign.
-	for _, vIn := range vPkt.Inputs {
-		key := vIn.Asset().TapCommitmentKey()
-		assetCommitment, ok := passiveCommitments[key]
-		if !ok {
+	for _, vPkt := range vPackets {
+		for _, vIn := range vPkt.Inputs {
+			key := vIn.Asset().TapCommitmentKey()
+			assetCommitment, ok := passiveCommitments[key]
+			if !ok {
+				continue
+			}
+
+			err := removeAsset(assetCommitment, vIn.Asset(), key)
+			if err != nil {
+				return nil, fmt.Errorf("unable to "+
+					"delete asset: %w", err)
+			}
+		}
+	}
+
+	return passiveCommitments, nil
+}
+
+// determinePassiveAssetAnchorOutput determines the best anchor output to attach
+// passive assets to. If no suitable output is found, a new anchor output is
+// created.
+func (f *AssetWallet) determinePassiveAssetAnchorOutput(ctx context.Context,
+	activePackets []*tappsbt.VPacket) (*keychain.KeyDescriptor, uint32,
+	error) {
+
+	var (
+		maxAnchorOutputIndex uint32
+		candidates           []*tappsbt.VOutput
+		candidateDescriptos  []*keychain.KeyDescriptor
+	)
+	for idx := range activePackets {
+		vPkt := activePackets[idx]
+
+		for _, vOut := range vPkt.Outputs {
+			anchorKeyDesc, err := vOut.AnchorKeyToDesc()
+			if err != nil {
+				// We can't determine the key descriptor for
+				// this output, so we'll skip it as it very
+				// likely doesn't belong to us then.
+				continue
+			}
+
+			// Ignore any anchor outputs that are not local to us.
+			if !f.cfg.KeyRing.IsLocalKey(ctx, anchorKeyDesc) {
```

Comment:
is this essentially checking if the anchor belongs to us?

Reply:
Correct! Since we'll want to re-anchor any passive assets back to our wallet.